### PR TITLE
Update to Vue 3 and add Typescript definition

### PR DESCRIPTION
**Pull request by Trinovantes**

### CODE REVIEW

**Comment by joshgoebel** [comment]
Wow.  Thanks.

I assume Vue 2 isn't going anywhere though, so now we have to figure out how to support both... 

- Do we just need a new branch and Vue 2 becomes "deprecated"?
- Can this be made compatible with both 2 and 3?
- Should both be provided side by side (as sep source files) in this repo and maintained?
- Two repos?

How are most other Vue plugins handling this?

**Comment by Trinovantes** [comment]
Most plugins just use a major version release bump so npm/yarn updates won't introduce breaking changes e.g. Vuex 3 supports Vue 2, Vuex 4 supports Vue 3 (awkward numbering)

Vue 2 users should also be able to use this plugin if they use `@vue/composition-api` that basically pollyfills the Vue 3 API into Vue 2

File: src/highlight-vue.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { ref, h, computed, defineComponent, Plugin } from 'vue'
+import hljs from 'highlight.js/lib/core'
+
+function escapeHtml(value: string): string {
```
Comment (joshgoebel) [inline src/highlight-vue.ts:4]: Curious, what was the issue with this utility living in `lib/utils`?

Comment (Trinovantes) [reply src/highlight-vue.ts:4]: No real reason, just did it on a whim. Want me to put it back?

Comment (Trinovantes) [reply src/highlight-vue.ts:4]: I also just noticed vue provides an `escapeHtml` function as well https://github.com/vuejs/vue-next/blob/master/packages/shared/src/escapeHtml.ts

Comment (joshgoebel) [reply src/highlight-vue.ts:4]: Since it escapes all 5 I think we could use that if it lightens the payload a bit.

When is Vue 3 going to be released?

Comment (Trinovantes) [reply src/highlight-vue.ts:4]: It was released Sept last year

Comment (joshgoebel) [reply src/highlight-vue.ts]: Ah, they just plan to keep their `next` tagging in NPM forever then I suppose? :-)

Comment (Trinovantes) [reply src/highlight-vue.ts]: Pretty much yea

**Comment by joshgoebel** [comment]
Can this please be rebased to pull in the small change from https://github.com/highlightjs/vue-plugin/pull/4?

Sorry for the delay.  After that I'll find the time to get this merged and make a new release.

**Comment by Trinovantes** [comment]
Rebased with current `HEAD`

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**